frontend/src/employee-frontend/components/unit/tab-calendar/discussion-surveys/times-calendar/CalendarEventTimeReservation.tsx

```diff
@@ -0,0 +1,363 @@
+                }}
+              />
+            </FixedSpaceRow>
+          )}
+
+          <Gap size="X5L" />
+          <FixedSpaceRow justifyContent="space-between">
+            <InlineButton
+              text={i18n.common.remove}
+              onClick={() => {
+                if (savedChild) {
+                  setDeleteConfirmModalVisible(true)
+                } else {
+                  deleteCalendarEventTime({
+                    eventId: eventData.id,
+                    id: eventTime.id
+                  })
+                    .then(() => onClose(true))
+                    .catch(() =>
+                      setErrorMessage({
+                        type: 'error',
+                        title: t.reservationModal.deleteError,
+                        resolveLabel: i18n.common.ok
+                      })
+                    )
+                }
+              }}
+              data-qa="delete-reservation-button"
+              icon={faTrash}
+            />
+
+            <FixedSpaceRow justifyContent="flex-end" spacing="s">
+              <Button
+                onClick={cancelChanges}
+                data-qa="cancel"
+                text={i18n.common.cancel}
+              />
+              <MutateButton
+                primary
+                onClick={() => ({
+                  body: {
+                    calendarEventTimeId: eventTime.id,
+                    childId: reservationChild.value().childId
+                  },
+                  eventId: eventData.id
+                })}
+                onSuccess={() => onClose(true)}
+                text={i18n.unit.calendar.events.edit.saveChanges}
+                mutation={setCalendarEventTimeReservationMutation}
+                data-qa="submit-reservation-button"
+                disabled={reservationChild.state.childId === eventTime.childId}
+              />
+            </FixedSpaceRow>
+          </FixedSpaceRow>
+          <Gap size="L" />
+        </BaseModal>
+
+        {cancelConfirmModalVisible && (
+          <InfoModal
+            type="warning"
+            title={t.cancelConfirmation.title}
+            icon={faQuestion}
+            reject={{
+              action: () => setCancelConfirmModalVisible(false),
+              label: t.cancelConfirmation.cancelButton
+            }}
+            resolve={{
+              action: () => {
+                setCancelConfirmModalVisible(false)
+                onClose(false)
+              },
+              label: t.cancelConfirmation.continueButton
+            }}
+            text={t.cancelConfirmation.text}
+          />
+        )}
+        {deleteConfirmModalVisible && (
+          <InfoModal
+            type="warning"
+            title={t.reservationModal.deleteConfirmation.title}
+            icon={faQuestion}
+            reject={{
+              action: () => setDeleteConfirmModalVisible(false),
+              label: t.reservationModal.deleteConfirmation.cancelButton
+            }}
```

Comment:
Consider using ConfirmedMutation component here. It is meant for the common use-case of "click button -> show confirmation modal -> if user confirms the run the mutation" so if these two don't include anything special, you would get the same functionality with less code.

Reply:
Modal behavior here has a "special" detail: confirmation is prompted conditionally based on the change detection to the reservation information, e.g. confirmation on deletion is only prompted when deleting a reserved event time. As I understand, `ConfirmedMutation` will always prompt for confirmation and would result in less cognizant behavior.